### PR TITLE
scheduler/add: interface check on nodeaffinity

### PR DESCRIPTION
**Pull request by sanposhiho**
Hello team! 👋

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

```
var _ framework.ScorePlugin = &NodeAffinity{}
```

The above is a way to make it clear to the developer which interface the structure satisfies and to have the compiler check it.

There are some interfaces that NodeAffinity satisfies, but are not checked by this syntax. This puts the developer at the mercy of the code they are reading. So fix them.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go [hunk 1]
```diff
@@ -36,8 +36,11 @@ type NodeAffinity struct {
 	addedPrefSchedTerms *nodeaffinity.PreferredSchedulingTerms
 }
 
+var _ framework.PreFilterPlugin = &NodeAffinity{}
 var _ framework.FilterPlugin = &NodeAffinity{}
+var _ framework.PreScorePlugin = &NodeAffinity{}
 var _ framework.ScorePlugin = &NodeAffinity{}
+var _ framework.ScoreExtensions = &NodeAffinity{}
```
Comment (chendave) [inline pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go:43]: Normally, the check is only added for extension point, should us add this to other plugin like this? e.g.
https://github.com/kubernetes/kubernetes/blob/d8f9e4587ac1265efd723bce74ae6a39576f2d58/pkg/scheduler/framework/plugins/podtopologyspread/scoring.go#L254-L256

Comment (ahg-g) [reply pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go:43]: no need for this one, it is part of ScorePlugin interface.

Comment (sanposhiho) [reply pkg/scheduler/framework/plugins/nodeaffinity/node_affinity.go:43]: Okey, thanks
revert this change.

**Pull request merged**